stock_available/views/product_product_view.xml

```diff
@@ -15,7 +15,7 @@
                 <button
                     type="object"
                     name="action_open_quants"
-                    invisible="type not in ('product', 'consu')"
+                    invisible="type != 'consu'"
```

Comment:
Only products with is_storable will have quants, so you can check this attribute (ref: https://github.com/odoo/odoo/blob/686628f55f6e/addons/stock/models/stock_quant.py#L595-L598)